### PR TITLE
Entity correlation

### PR DESCRIPTION
**Pull request by krzlabrdx**

### CODE REVIEW

File: src/RadixDlt.NetworkGateway.Abstractions/Model/EntityRelationship.cs [hunk 1]
```diff
@@ -62,29 +62,29 @@
  * permissions under this License.
  */
 
-using Microsoft.EntityFrameworkCore.Migrations;
+namespace RadixDlt.NetworkGateway.Abstractions.Model;
 
-#nullable disable
-
-namespace RadixDlt.NetworkGateway.PostgresIntegration.Migrations
+// TODO change to enum
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.Abstractions/Model/EntityRelationship.cs:67]: hmm I'm not sure, maintaining string might be easier. Also I'd make it more descriptive by nesting classes i.e
```
public static class EntityRelationship
{
    public static class Component
    {
        public const string Package
    }

    public static class Validator 
    {
    }

... etc
}
```

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/RelationshipProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        }
+
+        if (substateData is CoreModel.TypeInfoModuleFieldTypeInfoSubstate typeInfoSubstate)
+        {
+            switch (typeInfoSubstate.Value.Details)
+            {
+                case CoreModel.ObjectTypeInfoDetails objectDetails:
+                    referencedEntity.PostResolveConfigure((ComponentEntity e) =>
+                    {
+                        e.AssignedModuleIds = objectDetails.ModuleVersions
```
Comment (PawelPawelec-RDX) [inline src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/RelationshipProcessor.cs:103]: it's not related to relationships. maybe we have to process same substate in multiple places to keep it clean.

File: src/RadixDlt.NetworkGateway.Abstractions/Model/EntityRelationship.cs (hunk 1 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.Abstractions/Model/EntityRelationship.cs]: Had we decided to go with strings that would be no doubt the way to go. However I have rather strong objections here. We should go with enum as
- those relationships are well-defined and finite; we can always add new ones if needed,
- enums are lighter and easier to work with

File: src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/RelationshipProcessor.cs (hunk 2 above)
Comment (krzlabrdx) [reply src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/RelationshipProcessor.cs]: Ah, I noticed that myself and wanted to change that before pushing... obviously I must have forgotten about about that before I pushed.

**Review by PawelPawelec-RDX — APPROVED** [review]

**Review by PawelPawelec-RDX — APPROVED** [review]

**Pull request merged**